lib/Backend/GlobOpt.cpp

```diff
@@ -9585,6 +9584,32 @@ GlobOpt::TypeSpecializeInlineBuiltInBinary(IR::Instr **pInstr, Value *src1Val, V
             break;
         }
 
+        case Js::OpCode::InlineMathPow:
+        {
+#ifndef _M_ARM
+            if (src2Val->GetValueInfo()->IsLikelyInt())
+            {
+                bool lossy = false;
+
+                IR::Opnd* src1 = instr->GetSrc1();
+                this->ToFloat64(instr, src1, this->currentBlock, src1Val, nullptr, IR::BailOutPrimitiveButString);
+
+                IR::Opnd* src2 = instr->GetSrc2();
+                this->ToInt32(instr, src2, this->currentBlock, src2Val, nullptr, lossy);
+
+                IR::Opnd* dst = instr->GetDst();
+                this->ToFloat64Dst(instr, dst->AsRegOpnd(), this->currentBlock);
+            }
```

Comment:
Good catch.

---
In reply to: [57645551](https://github.com/Microsoft/ChakraCore/pull/644#discussion_r57645551) [](ancestors = 57645551)